### PR TITLE
Add hyperwasm bindings for `calcSpotPriceAfterLong`, `calcSpotPriceAfterShort`, `calcCloseShort`, `calcCloseLong`, `getOpenLongCurveFees`, and `getOpenShortCurveFees`

### PR DESCRIPTION
**Pull request by DannyDelott**

### CODE REVIEW

File: crates/hyperdrive-wasm/src/long.rs [hunk 1]
```diff
@@ -66,3 +66,85 @@ pub fn getMaxLong(
...
+    poolConfig: &JsPoolConfig,
+    baseAmount: String,
+) -> String {
+    set_panic_hook();
+    let state = State {
+        info: poolInfo.into(),
+        config: poolConfig.into(),
+    };
+
+    let _baseAmount: FixedPoint = FixedPoint::from(U256::from_dec_str(&baseAmount).unwrap());
```
Comment (ryangoree) [inline crates/hyperdrive-wasm/src/long.rs:89]: Nit: Some of the existing functions have these type annotations on them, but they're unnecessary. The type will be inferred.
```suggestion
    let _baseAmount = FixedPoint::from(U256::from_dec_str(&baseAmount).unwrap());
```

File: crates/hyperdrive-wasm/src/long.rs [hunk 2]
```diff
@@ -66,3 +66,85 @@ pub fn getMaxLong(
...
+    poolInfo: &JsPoolInfo,
+    poolConfig: &JsPoolConfig,
+    baseAmount: String,
+) -> String {
+    set_panic_hook();
+    let state = State {
+        info: poolInfo.into(),
+        config: poolConfig.into(),
+    };
+    let _baseAmount: FixedPoint = FixedPoint::from(U256::from_dec_str(&baseAmount).unwrap());
```
Comment (ryangoree) [inline crates/hyperdrive-wasm/src/long.rs:115]:
```suggestion
    let _baseAmount = FixedPoint::from(U256::from_dec_str(&baseAmount).unwrap());
```

File: crates/hyperdrive-wasm/src/short.rs [hunk 3]
```diff
@@ -88,3 +88,97 @@ pub fn getMaxShort(
...
+    shortAmount: String,
+    spotPrice: String,
+) -> String {
+    set_panic_hook();
+    let state = State {
+        info: poolInfo.into(),
+        config: poolConfig.into(),
+    };
+    let _shortAmount: FixedPoint = FixedPoint::from(U256::from_dec_str(&shortAmount).unwrap());
+    let _spotPrice: FixedPoint = FixedPoint::from(U256::from_dec_str(&spotPrice).unwrap());
```
Comment (ryangoree) [inline crates/hyperdrive-wasm/src/short.rs:114]:
```suggestion
    let _shortAmount = FixedPoint::from(U256::from_dec_str(&shortAmount).unwrap());
    let _spotPrice = FixedPoint::from(U256::from_dec_str(&spotPrice).unwrap());
```

File: crates/hyperdrive-wasm/src/short.rs [hunk 4]
```diff
@@ -88,3 +88,97 @@ pub fn getMaxShort(
...
+    poolInfo: &JsPoolInfo,
+    poolConfig: &JsPoolConfig,
+    bondAmount: String,
+) -> String {
+    set_panic_hook();
+    let state = State {
+        info: poolInfo.into(),
+        config: poolConfig.into(),
+    };
+    let _bondAmount: FixedPoint = FixedPoint::from(U256::from_dec_str(&bondAmount).unwrap());
```
Comment (ryangoree) [inline crates/hyperdrive-wasm/src/short.rs:140]:
```suggestion
    let _bondAmount = FixedPoint::from(U256::from_dec_str(&bondAmount).unwrap());
```

**Review by ryangoree — APPROVED** [review]

File: crates/hyperdrive-wasm/src/long.rs (hunk 1 above)
Comment (DannyDelott) [reply crates/hyperdrive-wasm/src/long.rs]: Awesome, good to know!

**Pull request merged**